### PR TITLE
List Jobs executed in a range of time

### PR DESCRIPTION
**Pull request by kuisathaverat**
@reviewbybees

### CODE REVIEW

File: scriptler/jobExecutedInDateRange.groovy [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+ /*** BEGIN META {
+  "name" : "List executed Jobs",
+  "comment" : "List Jobs executed in a range of time",
+  "core": "1.609",
+  "authors" : [
+  { name : "Kuisathaverat" }
+  ]
+  } END META**/
+
```
Comment (imod) [inline scriptler/jobExecutedInDateRange.groovy]: `BEGIN META` should only be once in the file

**Comment by imod** [comment]
waiting for @reviewbybees ....

File: scriptler/jobExecutedInDateRange.groovy [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*** BEGIN META {
+ "name" : "List executed Jobs",
+ "comment" : "List Jobs executed in a range of time",
+ "core": "1.609",
+ "authors" : [
+ { name : "Kuisathaverat" }
+ ]
+ } END META**/
+
+ import jenkins.model.*
```
Comment (Evildethow) [inline scriptler/jobExecutedInDateRange.groovy]: Import not needed (pre-imported)

File: scriptler/jobExecutedInDateRange.groovy [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ import java.util.Date
```
Comment (Evildethow) [inline scriptler/jobExecutedInDateRange.groovy]: Unused import. Especially if you remove the unused date below `Date now = new Date()`

File: scriptler/jobExecutedInDateRange.groovy [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+ jenkins = Jenkins.instance
+
+ int count = 0
```
Comment (Evildethow) [inline scriptler/jobExecutedInDateRange.groovy]: Assignment `count` is never used

File: scriptler/jobExecutedInDateRange.groovy [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ Date now = new Date()
+ Calendar after = Calendar.getInstance()
+ Calendar before = Calendar.getInstance()
+ //set(int year, int month, int date, int hourOfDay, int minute,int second)
+ after.set(2016,1,5,0,0,0)
+ before.set(2016,1,23,0,0,0)
+ println "Jobs executed between " + after.getTime() + " - " + before.getTime()
```
Comment (Evildethow) [inline scriptler/jobExecutedInDateRange.groovy]:
```
println "Jobs executed between " + after.getTime() + " - " + before.getTime()
```

should be

```
println "Jobs executed between " + before.getTime() + " - " + after.getTime()
```

?

File: scriptler/jobExecutedInDateRange.groovy [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+ for (job in jenkins.getAllItems(Job.class)) {
```
Comment (Evildethow) [inline scriptler/jobExecutedInDateRange.groovy]: Can be re-written to:

```
jenkins.getAllItems(Job.class).each { Job job ->
  job.getBuilds()
      .findAll { (it.getTimestamp()?.before(before) && it.getTimestamp()?.after(after)) }
      .each { Run run ->
    println "${run.getResult()} ${job.name} ${job.getLastBuild().getTime()}"
  }
}
```

Just a comment.

**Comment by Evildethow** [comment]
🐝

**Comment by Evildethow** [comment]
@kuisathaverat Maybe `rebase` after you are done if you are going to make `fixup` commits. Doesn't bother me either way to be honest :) but I know some like their git log shiny and clean.

**Comment by Dohbedoh** [comment]
🐝

**Comment by kuisathaverat** [comment]
@reviewbybees done

**Pull request merged**